### ISSUE TITLE
Make ApiResource shorter

### ISSUE THREAD

**Issue by tourze** [api-platform/core#5076]
Sometime we need to write a long code like :
```php
#[ApiResource(
    routePrefix: '/live-chat',
    normalizationContext: ['groups' => ['restful_read']],
    denormalizationContext: ['groups' => ['restful_write']],
    openapiContext: ['tags' => ['LiveChat']],
    order: ['id' => 'DESC'],
    security: 'is_granted("ROLE_OPERATOR")'
)]
```

It is too long for person to read.
Can we just create a new Attribute class, `class RestfulApi extends ApiResource`, and move some normal logic into consturctor?
Then we just add code like this:

```php
#[RestfulApi(
    routePrefix: '/live-chat',
)]
```

Related code: 
https://github.com/api-platform/core/blob/1a3ae348dfd35d6d4117a7fcecaf1ecd1fe1e4ea/src/Metadata/Resource/Factory/AttributesResourceNameCollectionFactory.php#L60

**Comment by yobrx** [comment]
@dunglas https://github.com/api-platform/core/pull/5175

My first PR on Api Platform :pray:

**Comment by yobrx** [comment]
It's fixed on 3.0.4 last upgrade
